### ISSUE TITLE
Rectangular Tab border instead of circular

### ISSUE THREAD

**Issue by iseeghosts** [microsoft/terminal#7213]
# Rectangular border like Microsoft Edge Legacy

Current Look
![image](https://user-images.githubusercontent.com/34771316/89641440-470bb180-d8cf-11ea-84cb-ccd2c1085e2b.png)
What should it look like
![image](https://user-images.githubusercontent.com/34771316/89641489-63a7e980-d8cf-11ea-8555-4ab942f7df7f.png)

**Comment by zadjii-msft** [comment]
Thanks for the suggestion! This is something that's already being discussed in #5772, but I'll tag this up as a sub-task of #3327. Thanks!

**Comment by zadjii-msft** [comment]
Notes:

* Overriding the tab corner radius ON THE TOP seems fairly trivial. Just set the `OverlayCornerRadius` resource to whatever. 
* Changing the bottom seems **H**ard. Seems like there's no easy way to hide `(Left|Right)RadiusRenderArc.Visibility` on the TabViewItem itself. So you end up with this weird square on top, rounded into the terminal bottom that looks awful.
* How would we express "I want the bottom tab corners to round _inwards_, so they look like buttons? 
  - I'm thinking we'd want the bottom corners to accept a positive number as the current outwards radius, and a negative number to be the inwards radius. 
  - We'd need a `TabViewItemCornerRadiusConverter` to convert the negative radius to something sensible
  - We'd need `TabViewItemCornerArcVisibilityConverter` that collapses the bottom arcs when the corner radius set by the user is negative

**Comment by zadjii-msft** [comment]
From discussion with WinUI folks:

> I think you would have to retemplate the TabViewItem component.
> 
> I would try the following:  Apply a custom Style to TabViewItem without the LeftRadiusRenderArc, RightRadiusRenderArc and SelectedBackgroundPath elements. You get this which is not enough:
> 
> ![image](https://user-images.githubusercontent.com/18356694/174143648-93727aa8-ae60-452b-b871-4a296af3ab5a.png)
> 
> You need to extend the vertical lines down. Maybe TabViewItemSeparatorMargin affects that - I'm not sure.
> 
> You may have to replace the SelectedBackgroundPath with your own element, because unfortunately its shape is hardcoded in TabViewItem::UpdateTabGeometry(). This is pretty bad I know.
> 
> The key problem is this hardcoded line in \dev\TabView\TabViewItem.cpp
> ```c++
> auto data = L"<Geometry xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>F1 M0,%f a 4,4 0 0 0 4,-4 L 4,%f a %f,%f 0 0 1 %f,-%f l %f,0 a %f,%f 0 0 1 %f,%f l 0,%f a 4,4 0 0 0 4,4 Z</Geometry>";
> ```
>
> I think you'd have to write the same kind of code without the 2 bottom curves.  Then call your own version of UpdateTabGeometry when the size changes like we do in TabViewItem::OnSizeChanged.


We'd need to make a specific request on WinUI to allow this to be changed in the tab view.

**Comment by dansmor7** [comment]
If you are willing to give up the two corner overlays, then customizing the entire tab can be easily done through TabContainer, without having to update the path through code.

Otherwise, I'd suggest using a single positive corner radius and another property for the tab attaching to the bottom or not.
This is for the eventuality that TabView will support different orientations in the future (+ negative radius doesn't make sense for the other two corners).
For the current design, it would look like this:
```XAML
<TabViewItem CornerRadius="8,8,4,4" IsAttached="True">
```

**Comment by zadjii-msft** [comment]
That seems like a pretty sensible approach to me - we should make sure to capture that in the feature request on the WinUI team (whenever we get around to filing it)

**Comment by dansmor7** [comment]
The problem is that it would be fairly difficult to achieve. The code would need to render the entire tab, taking into account border thickness and corner radius, effectively recreating the Xaml Border/Panel but with two of the corners being inverted.

It would be much easier to drop the overlays for now, which would allow full customization, including border thickness.
The default design would need to change to match the latest fluent design, so the tabs would end up looking like either of these:
![image](https://user-images.githubusercontent.com/101892345/177384350-861e77b8-f6e6-43a9-882a-63bcd216d03f.png)

**Comment by zadjii-msft** [comment]
Maybe I've been overcomplicating by trying to dedupe this with https://github.com/microsoft/terminal/issues/3327#issuecomment-765493313. There's basically two sides to this spectrum
* I want the tabs to be rectangular, without curved corners
* I want the tabs to be buttons / pills. That should both include pills (floating "tabs", aka buttons), but also just like, rectangular buttons (like [this](https://media.gcflearnfree.org/ctassets/topics/14/wn_switchwindows.gif))

**Comment by zadjii-msft** [comment]
WinUI thread: https://github.com/microsoft/microsoft-ui-xaml/issues/7032

**Comment by riotrah** [comment]
Hi just to be clear - the work referenced being done in this issue is to _allow_ end-users to make such OS/1st party UI-disruptive changes _at their own peril_, right? Or at a maximum, perhaps check if they're on win10 vs 11 or something? 

_Most certainly_ a linguistic/tonal/inference thing on my part, but if anyone is _actually_ demanding regressing back to right angles _again_ in 2022 as _"the"_ WT design, on the basis of an obviously subjective and almost certainly minority-held premise of "rounded ugly" then I'll eat my shoe 🙄

**Comment by zadjii-msft** [comment]
Yes, to be totally clear - this would be a setting as a part of Themes. This would absolutely not be something we roll out by default to all users. It'd instead be a knob users could set _if they so choose_.

**Comment by Shomnipotence** [comment]
![image](https://github.com/microsoft/terminal/assets/6630660/152b3001-d3b8-41cf-9a90-fa2bdca8f96a)
